sources/json_parser.c

```diff
@@ -524,8 +524,13 @@ bool json_deserialize_file_management_parameter(char* buffer, size_t buffer_size
                 return false;
             }
 
-            file_management_parameter_set_filename(parameter, value_buffer);
-            i++;
+            if (strlen(value_buffer) > FILE_MANAGEMENT_FILE_NAME_SIZE) {
+                /* Leave file name array empty */
+                return true;
```

Comment:
Why true?

Reply:
parser vise it is legal state, nothing breaks, and I based check on the array length so I need true to start handling:
```
  const size_t num_deserialized_parameter = parser_deserialize_file_management_parameter(
      &ctx->parser, (char*)payload, (size_t)payload_len, &file_management_parameter);
  if (num_deserialized_parameter != 0) {
      _handle_file_management_parameter(&ctx->file_management, &file_management_parameter);
```
NOTE: parser that you looking at is called by parser_deserialize_file_management_parameter